src/ecrc-api/src/main/java/ca/bc/gov/open/ecrc/service/EcrcPaymentServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+package ca.bc.gov.open.ecrc.service;
+
+import javax.annotation.PostConstruct;
+
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import ca.bc.gov.open.ecrc.configuration.EcrcProperties;
+import ca.bc.gov.open.ecrc.exception.EcrcExceptionConstants;
+import ca.bc.gov.open.ecrc.exception.EcrcServiceException;
+import ca.bc.gov.open.ecrc.exception.WebServiceStatusCodes;
+import ca.bc.gov.open.ecrc.model.RequestPaymentService;
+import ca.bc.gov.open.ecrc.model.ResponsePaymentService;
+import reactor.core.publisher.Mono;
+
+/**
+ *
+ * Payment Service Implementation class.
+ *
+ * @author sivakaruna
+ *
+ */
+@Service
+@Configuration
+@EnableConfigurationProperties(EcrcProperties.class)
+public class EcrcPaymentServiceImpl implements EcrcPaymentService {
+
+	@Autowired
+	private EcrcProperties ecrcProps;
+
+	@Autowired
+	ObjectMapper objectMapper;
+
+	private WebClient webClient = null;
+
+	private final Logger logger = LoggerFactory.getLogger(EcrcPaymentServiceImpl.class);
+
+	@PostConstruct
+	public void InitService() {
+
+		this.webClient = WebClient.create(ecrcProps.getPaymentUrl());
```

Comment:
One change we will need to make is this will use basicauth in the future.

Reply:
Thanks, Taylor. I will make a few updates to this PR to use constants as well